### ISSUE TITLE
Upgrade to React 16 - Region without empty DIV

### ISSUE THREAD

**Issue by fahadsuhaib** [frintjs/frint#365]
The below code block in region.js is not required in React 16, it could be simply returned as a listForRendering.map, React 16 considers array rendering as well and that helps in minimizing bloated HTML.

```
return _react2.default.createElement(
        'div',
        null,
        listForRendering.map(function (item) {
          var Component = item.Component,
              name = item.name;


          return _react2.default.createElement(Component, { key: 'app-' + name });
        })
      );
```

**Comment by fahad19** [comment]
@fahadsuhaib: thanks Fahad! could you kindly send a PR for this? :)

**Comment by mAiNiNfEcTiOn** [comment]
@fahad19 I'd like to point out that - contrarily to what is in this example code - the `<div>` on the `frint-react`'s Region contains a reference to a variable `className` provided via props.

By removing the extra div, you might be breaking functionality on other usages.

Here's the place where it's used: https://github.com/frintjs/frint/blob/master/packages/frint-react/src/components/Region.js#L63

**Comment by mAiNiNfEcTiOn** [comment]
Yeah, I was preparing a Pull Request for this, but this is a breaking change... Do we really want this?

I see the wrapping `<div>` as a definition of the Region (or its boundaries). Otherwise, if we return fragments what'll happen is something that was like this:

``html
<main>
  <div class="my-custom-region-class">
    <App 1 />
    <App 2 />
    <App 3 />
    <App 4 />
  </div>
</main>
```

become:
```html
<main>
  <App 1 />
  <App 2 />
  <App 3 />
  <App 4 />
</main>
```

Which means:

* The DOM stops having the concept of Region, just knows the apps.
* The support for 'my-custom-region-class' is gone.
* The existing usages of the className attribute would stop working.


Also, company-wise, this is a specific feature of R16, which _this repo_ is in. :wink:

**Comment by fahad19** [comment]
@mAiNiNfEcTiOn: I agree. this does change the behaviour.

**Comment by fahadsuhaib** [comment]
Conceptually, regions are place-holders so why should it have its own element? That should be the child component that should own the styling right?

**Comment by fahadsuhaib** [comment]
@mAiNiNfEcTiOn regarding your 3 points, I would still suggest that can be handled by the child component that is loads in the region, any update to Region should be handled via React.

**Comment by mAiNiNfEcTiOn** [comment]
@fahadsuhaib I understand your point of view. However, what I'm trying to say is that this is a breaking change.

Users that have been using the `Region` component are expecting that a `<div>` element comes out when rendering AND that they can pass a className as an attribute to that component and it will be used in the wrapper element rendered.

E.g.:

```
<style>
  .my-custom-class {
    border: solid 1 px black;
    border-radius: 5px;
  }
</style>

<Region className="my-custom-class" name="my-custom-name"/>
```

In the current state, this would output a `<div class="my-custom-class"><!-- Frint Apps here --></div>`

With the changes proposed, it would be just `<!-- Frint Apps here -->`.

If I update to the new way, the styles on the `<style>` would not be used anymore.


Last but not the least, I'm not against implementing it, but that is a breaking change which implies a new major. Also it means that FrintJS requires React 16 as a minimum. That means that for _deprecation_ and/or _migration_ periods you don't support previous versions of React.

Bold move, no?

**Comment by fahadsuhaib** [comment]
OK, could we include a props "useR16=true/false" that is false by default and when true it uses R16 way?

**Comment by fahad19** [comment]
we have some React v16 specific features planned for the future here: https://github.com/frintjs/frint/issues/334

until they are done, the best that I can say about this Issue right now is we can put it on hold, and come back to it later.
